source/DurationPicker.js

```diff
@@ -0,0 +1,733 @@
+			if(this.value !== '') {
+				this.valueChanged();
+			} else {
+				this.noneTextChanged();
+			}
+		},
+
+		/**
+		* creating the hour, minute and second picker components
+		*
+		* @private
+		*/
+		setupPickers: function () {
+			var len, idx, item;
+			len = this.templateArray.length;
+			if (len) {
+				for (idx = 0; idx < len; idx++) {
+					item = this.templateArray[idx];
+					switch (item) {
+					case 'h':
+						this.createComponent(
+							{name: 'hourPicker', classes: 'moon-date-picker-wrap', components:[
+								{kind: 'moon.IntegerPicker', name: 'hour', min: 0, max: 23, wrap: true, value: this.valueArray.hour, disabled: this.countdown},
+								{name: 'hourLabel', content: this.hourText, classes: 'moon-date-picker-label moon-divider-text'}
+							]}
+						);
+						break;
+					case 'm':
+						this.createComponent(
+							{name: 'minutePicker', classes: 'moon-date-picker-wrap', components:[
+								{kind: 'moon.IntegerPicker', name: 'minute', classes: 'moon-date-picker-field', min: 0, max: 59, wrap: true, value: this.valueArray.minute, disabled: this.countdown},
+								{name: 'minuteLabel', content: this.minuteText, classes: 'moon-date-picker-label moon-divider-text'}
+							]}
+						);
+						break;
+					case 's':
+						this.createComponent(
+							{name: 'secondPicker', classes: 'moon-date-picker-wrap', components:[
+								{kind: 'moon.IntegerPicker', name: 'second', classes: 'moon-date-picker-field', min: 0, max: 59, wrap: true, value: this.valueArray.second, disabled: this.countdown},
+								{name: 'secondLabel', content: this.secondText, classes: 'moon-date-picker-label moon-divider-text'}
+							]}
+						);
+						break;
+					}
+				}
+			} else {
+				this.createComponent(
+					{name: 'hourPicker', classes: 'moon-date-picker-wrap', components:[
+						{kind: 'moon.IntegerPicker', name: 'hour', min: 0, max: 23, wrap: true, value: this.valueArray.hour, disabled: this.countdown},
+						{name: 'hourLabel', content: this.hourText, classes: 'moon-date-picker-label moon-divider-text'}
+					]}
+				);
+				this.createComponent(
+					{name: 'minutePicker', classes: 'moon-date-picker-wrap', components:[
+						{kind: 'moon.IntegerPicker', name: 'minute', classes: 'moon-date-picker-field', min: 0, max: 59, wrap: true, value: this.valueArray.minute, disabled: this.countdown},
+						{name: 'minuteLabel', content: this.minuteText, classes: 'moon-date-picker-label moon-divider-text'}
+					]}
+				);
+				this.createComponent(
+					{name: 'secondPicker', classes: 'moon-date-picker-wrap', components:[
+						{kind: 'moon.IntegerPicker', name: 'second', classes: 'moon-date-picker-field', min: 0, max: 59, wrap: true, value: this.valueArray.second, disabled: this.countdownSS},
+						{name: 'secondLabel', content: this.secondText, classes: 'moon-date-picker-label moon-divider-text'}
+					]}
+				);
+			}
+			this.render();
+		},
+
+		/**
+		* @private
+		*/
+		currentTemplate: function () {
```

Comment:
On the nitpicky side again, but this sounds more like a property name than a method - how about something like 'initTemplate` or `parseTemplate`?